makefile

```diff
@@ -115,12 +129,25 @@ ifeq "$(wildcard $(JAVA_HOME) )" ""
 	$(eval JAVA_HOME = $(shell $(JAVA) -jar $(JYTHON) -c "from java.lang import System; print System.getProperty('java.home')[:-4]"))
 endif
 
+# assume you want debug if you are just compiling one portion of the codebase
+libJyNI:  CFLAGS += -g
 libJyNI: $(JAVA_HOME) $(OBJECTS) JyNI-C/src/Python/dynload_shlib.o
```

Comment:
Interesting. I never really studied makefile syntax systematically. Does this really work as expected? I.e. with the double entry ``libJyNI:``

Reply:
Yes, oddly enough it does work as expected. It seems to simply run both the targets, it may be possible to do it slightly differently, I'll see if I can make it cleaner. (But I'm not fantastic with makefiles)

Reply:
I thought I had an idea, but it turns out I already tried it and it didn't work, unfortunately I've yet to find a better approach. It seems this is a thing called [Target-specific Variable Values](https://www.gnu.org/software/make/manual/make.html#Target_002dspecific). I know it works in gcc, but I don't know for clang or osx.

Reply:
https://www.gnu.org/software/make/manual/html_node/Multiple-Rules.html seems to state that it actually would *not* work as we expect:

> There can only be one recipe to be executed for a file. If more than one rule gives a recipe for the same file, make uses the last one given and prints an error message.

Also see https://docs.oracle.com/cd/E19504-01/802-5880/make-24/index.html and https://www.gnu.org/software/make/manual/html_node/Double_002dColon.html#Double_002dColon.

I'd suggest to rename in terms of the original makefile ``libJyNI`` to ``_libJyNI`` *everywhere*. Then introduce a new ``libJyNI: _libJyNI _debug`` with also changing

```
debug: CFLAGS += -g
debug: all
```

to 

```
_debug: CFLAGS += -g
debug: all _debug
```

We can do the same for the loader but maybe changes in the loader are so rare that it is sufficient to let it just build everything then.